flyteadmin/pkg/manager/impl/execution_manager.go

```diff
@@ -862,10 +863,14 @@ func (m *ExecutionManager) launchExecutionAndPrepareModel(
 	})
 
 	if err != nil {
-		m.systemMetrics.PropellerFailures.Inc()
-		logger.Infof(ctx, "Failed to execute workflow %+v with execution id %+v and inputs %+v with err %v",
-			request, workflowExecutionID, executionInputs, err)
-		return nil, nil, err
+		if strings.Contains(err.Error(), "no node can run task") {
```

Comment:
instead of string parsing I suggest to introduce some kind of `RuntimeError` struct that will contain error details. We can type check against this type and it will denote that we should still save execution model to database, but with Failed phase and error details.